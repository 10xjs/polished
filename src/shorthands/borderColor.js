// @flow
import directionalProperty from '../helpers/directionalProperty'

function borderColor(...values: Array<string|null>) {
  return directionalProperty('border-color', ...values)
}

export default borderColor
