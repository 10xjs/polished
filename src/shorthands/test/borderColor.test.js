// @flow
import borderColor from '../borderColor'

describe('borderColor', function() {
  it('properly applies a value when passed only one', function() {
    expect(borderColor('12px')).toMatchSnapshot()
  })
  it('properly applies values when passed two', function() {
    expect(borderColor('12px', '24px')).toMatchSnapshot()
  })
  it('properly applies values when passed three', function() {
    expect(borderColor('12px', '24px', '36px')).toMatchSnapshot()
  })
  it('properly applies values when passed four', function() {
    expect(borderColor('12px', '24px', '36px', '48px')).toMatchSnapshot()
  })
})
