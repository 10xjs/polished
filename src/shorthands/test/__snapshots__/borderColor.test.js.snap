exports[`borderColor properly applies a value when passed only one 1`] = `
Object {
  "border-bottom-color": "12px",
  "border-left-color": "12px",
  "border-right-color": "12px",
  "border-top-color": "12px",
}
`;

exports[`borderColor properly applies values when passed four 1`] = `
Object {
  "border-bottom-color": "36px",
  "border-left-color": "48px",
  "border-right-color": "24px",
  "border-top-color": "12px",
}
`;

exports[`borderColor properly applies values when passed three 1`] = `
Object {
  "border-bottom-color": "36px",
  "border-left-color": "24px",
  "border-right-color": "24px",
  "border-top-color": "12px",
}
`;

exports[`borderColor properly applies values when passed two 1`] = `
Object {
  "border-bottom-color": "12px",
  "border-left-color": "24px",
  "border-right-color": "24px",
  "border-top-color": "12px",
}
`;
